### ISSUE TITLE
Crash (password request) OR call activity fails to stop OR call screen does not close after hanging up a call immediately

### ISSUE THREAD

**Issue by Rolf558** [WhisperSystems/Signal-Android#6348]
Hi,

If I make a call (both devices WebRTC enabled) and hang up immediately (before the recipient's phone rings) I often (but not always) get to Signal's login screen and have to enter the password again. I am not sure what happens. Perhaps Signal crashes and re-starts.

Found with build 3.31.0 for Android.

**Comment by moxie0** [comment]
Hey @Rolf558, thanks for the report but please follow the issue template and include a debug log if possible.

**Comment by Rolf558** [comment]
Hi @moxie0,

please have a look at
https://gist.github.com/9f5a33191c8391d6bae787ada75f6da6
and let me know if this one helps. 
(If not I could try to make a fresh cat.log directly after such a password request.)

**Comment by McLoo** [comment]
I twitched over a while over call and hang-up button. A finally got Signal crashed. 
https://gist.github.com/d55cb8015e5a93d425055c748032564a

Also sent the crash report (not sure if it really got sent...) 

Will do some more twitching there, as I observed another bug. The first two times I did the rapid call and hang-up the call activity was not closed. 
Then the other 7 to 10 attempts all was fine, until the crash.

**Comment by Rolf558** [comment]
> ...hang-up the call activity was not closed.

I experienced this bug several times too. I have adjusted the summary.

**Comment by cascheberg** [comment]
Possibly related: #6305

**Comment by Rolf558** [comment]
In build 3.31.3 I see the following behavior (a is new, b is unchanged):

a) After hanging up immediately, the  call screen does no more close. (I have to hit the back button to get rid of the call screen). (But I had no more crash in my 3 tests.)

b) After hanging up immediately [but perhaps a little bit alter than in a)] the problem that call activity is not stopped still persists with build 3.31.3.

**Comment by Rolf558** [comment]
Build 4.2.4 is still affected by these problems.

**Comment by Rolf558** [comment]
Signal 4.6.0 is still affected by these issues.

Is additional information required? I wonder why this bug report has no developer attention.

**Comment by Rolf558** [comment]
@moxie0 I would appreciate If would drop a note if I can do anything to support fixing this issues (without software development skills).

**Comment by Rolf558** [comment]
@moxie0 according to your request I have added a better description of the issue following the template. In addition I have added a cat.log.

**Bug description**

Signal 4.6.0 has introduce a confirmation dialogue when sending locations. If I try to confirm a location during bad network conditions (so that Signal fails to load the map) Signal crashes.

**Steps to reproduce**

- Call someone using Signal's built in WebRTC VoIP feature
- Immediately hit the hang-up button

**Actual result:**

Now we have either:
a) crash (password request because of app re-start)
b) call activity does not stop
c) call screen does not close

**Epected result:**

No crash. Call screen should close.

**Screenshots**

n/a

**Device info**

Device: LG G5 SE
Android version: 7.0
Signal version: 4.6.1

**Link to debug log**
https://gist.github.com/d4f21970266ee06f2c56033bfea18305
